### PR TITLE
SONARGRADL-10 Update to SQ Runner API 2.5

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 1]
```diff
@@ -38,32 +43,61 @@
  */
 public class SonarQubeTask extends DefaultTask {
 
-  private Map<String, Object> sonarProperties;
+    private static final Logger LOGGER = Logging.getLogger(SonarQubeTask.class);
+
+    public static final LogOutput LOG_OUTPUT = new LogOutput() {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 2]
```diff
@@ -38,32 +43,61 @@
...
 
-    return sonarProperties;
-  }
+    /**
+     * @return The String key/value pairs to be passed to the SonarQube Runner.
+     * {@code null} values are not permitted.
+     */
+    @Input
+    public Map<String, Object> getProperties() {
+        if (sonarProperties == null) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 3]
```diff
@@ -38,32 +43,61 @@
...
+            sonarProperties = Maps.newLinkedHashMap();
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 4]
```diff
@@ -38,32 +43,61 @@
  */
 public class SonarQubeTask extends DefaultTask {
 
-  private Map<String, Object> sonarProperties;
+    private static final Logger LOGGER = Logging.getLogger(SonarQubeTask.class);
+
+    public static final LogOutput LOG_OUTPUT = new LogOutput() {
+        @Override
+        public void log(String formattedMessage, Level level) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR The Cyclomatic Complexity of this method "log" is 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 5]
```diff
@@ -38,32 +43,61 @@
...
-  @TaskAction
-  public void run() {
-    Map<String, Object> properties = getProperties();
+    private Map<String, Object> sonarProperties;
 
-    Properties propertiesObject = new Properties();
-    propertiesObject.putAll(properties);
+    @TaskAction
+    public void run() {
+        Map<String, Object> properties = getProperties();
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 6]
```diff
@@ -38,32 +43,61 @@
  */
 public class SonarQubeTask extends DefaultTask {
 
-  private Map<String, Object> sonarProperties;
+    private static final Logger LOGGER = Logging.getLogger(SonarQubeTask.class);
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Make this line start at column 3. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 6 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 7]
```diff
@@ -38,6 +41,33 @@
  */
 public class SonarQubeTask extends DefaultTask {
 
+  private static final Logger LOGGER = Logging.getLogger(SonarQubeTask.class);
+
+  public static final LogOutput LOG_OUTPUT = new LogOutput() {
+    @Override
+    public void log(String formattedMessage, Level level) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR The Cyclomatic Complexity of this method "log" is 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/org/sonarqube/gradle/SonarQubeTask.java [hunk 8]
```diff
@@ -38,6 +41,33 @@
  */
 public class SonarQubeTask extends DefaultTask {
 
+  private static final Logger LOGGER = Logging.getLogger(SonarQubeTask.class);
+
+  public static final LogOutput LOG_OUTPUT = new LogOutput() {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubeTask.java]: MAJOR Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**